integration_test/model_nnet_demo_test.dart

```diff
@@ -0,0 +1,268 @@
+/// Model NNET test with demo dataset.
+//
+// Time-stamp: <Tuesday 2024-09-03 09:09:14 +1000 Graham Williams>
+//
+/// Copyright (C) 2024, Togaware Pty Ltd
+///
+/// Licensed under the GNU General Public License, Version 3 (the "License");
+///
+/// License: https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later
+// version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+// details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <https://www.gnu.org/licenses/>.
+///
+/// Authors: Zheyuan Xu
+
+library;
+
+// Group imports by dart, flutter, packages, local. Then alphabetically.
+
+import 'package:flutter/material.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'package:rattle/features/dataset/button.dart';
+import 'package:rattle/main.dart' as app;
+import 'package:rattle/widgets/image_page.dart';
+import 'package:rattle/widgets/text_page.dart';
+
+import 'utils/delays.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('NNET Model Demo Tree:', () {
+    testWidgets('default test.', (WidgetTester tester) async {
+      app.main();
+
+      await tester.pumpAndSettle();
+
+      await tester.pump(pause);
+
```

Comment:
PLEASE don't repeat code unnecessarily. We have already defined the openDemoDataset() util.